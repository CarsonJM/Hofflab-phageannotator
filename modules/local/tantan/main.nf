process TANTAN {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/tantan:49--h43eeafb_0':
        'biocontainers/tantan:49--h43eeafb_0' }"

    input:
    tuple val(meta) , path(fasta)

    output:
    tuple val(meta), path("${prefix}_tantan.tsv")   , emit: tantan
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    tantan \\
        ${fasta} \\
        -f 3 \\
        ${args} > ${prefix}_tantan.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tantan: \$(tantan --version 2>&1 | sed 's/^.*tantan //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_tantan.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tantan: \$(tantan --version 2>&1 | sed 's/^.*tantan //')
    END_VERSIONS
    """
}
