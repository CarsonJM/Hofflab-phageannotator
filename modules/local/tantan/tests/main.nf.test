nextflow_process {

    name "Test Process TANTAN"
    script "../main.nf"
    process "TANTAN"

    test("fasta.gz") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/fastas/lambda.fasta.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("fasta.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("${projectDir}/assets/test-datasets/fastas/lambda.fasta.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
