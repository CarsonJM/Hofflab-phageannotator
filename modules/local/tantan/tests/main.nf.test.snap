{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tantan.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ce1a51a6c238c0e82fb34f3e9f15a47f"
                ],
                "tantan": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tantan.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce1a51a6c238c0e82fb34f3e9f15a47f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T13:29:42.326188902"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tantan.tsv:md5,2c593b155de7b423450d215d19dba44f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ce1a51a6c238c0e82fb34f3e9f15a47f"
                ],
                "tantan": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tantan.tsv:md5,2c593b155de7b423450d215d19dba44f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce1a51a6c238c0e82fb34f3e9f15a47f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T13:29:23.182999965"
    }
}